packages/openapi-v3-types/src/openapi-v3-spec-types.ts

```diff
@@ -0,0 +1,401 @@
+  examples?: [ExampleObject | ReferenceObject];
+  example?: ExampleObject | ReferenceObject;
+  encoding?: EncodingObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface EncodingObject extends ISpecificationExtension {
+  [property: string]: EncodingPropertyObject | ExtensionValue;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface EncodingPropertyObject {
+  contentType?: string;
+  headers?: {[key: string]: HeaderObject | ReferenceObject};
+  style?: string;
+  explode?: boolean;
+  allowReserved?: boolean;
+  [key: string]: ExtensionValue;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface HeaderObject extends ParameterObject {}
+
+/**
+ * Describes a single operation parameter.
+ * A unique parameter is defined by a combination of a name and location.
+ * Specification
+ * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameter-object
+ */
+export interface ParameterObject extends ISpecificationExtension {
+  name: string;
+  in: ParameterLocation;
+  description?: string;
+  required?: boolean;
+  deprecated?: boolean;
+  allowEmptyValue?: boolean;
+
+  style?: ParameterStyle;
+  explode?: boolean;
+  allowReserved?: boolean;
+  schema?: SchemaObject | ReferenceObject;
+  examples?: {[param: string]: ExampleObject | ReferenceObject};
+  example?: ExtensionValue;
+  content?: ContentObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface ExampleObject {
+  summary?: string;
+  description?: string;
+  value?: ExtensionValue;
+  externalValue?: string;
+  [property: string]: ExtensionValue;
+}
+
+export interface ReferenceObject {
+  $ref: string;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface DiscriminatorObject {
+  propertyName: string;
+  mapping?: {[key: string]: string};
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface XMLObject extends ISpecificationExtension {
+  name?: string;
+  namespace?: string;
+  prefix?: string;
+  attribute?: boolean;
+  wrapped?: boolean;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface ExternalDocumentationObject extends ISpecificationExtension {
+  description?: string;
+  url: string;
+}
+
+/**
+ * Allow key(string) value extension specifications.
+ * Cannot constraint to "^x-" but can filter them later to access to them
+ */
+export interface ISpecificationExtension {
+  [extensionName: string]: ExtensionValue;
+}
+
+/**
+ * Maps names to a given type of values
```

Comment:
I find this one a bit confusing; is this just meant to be a `Map<T, U>`, where T is always a string?

Reply:
honestly I haven't investigated into those TS interfaces inherit from the old file...the scope of task is upgrading openapi specific spec, so I didn't spend too much effort on the pure TS interfaces.

`Map<T, U>` is `Map` a built-in interface provided by TS?

Reply:
Technically, maps are introduced by ES6 and there are corresponding types available for use in TS that describe them. I'm not super concerned about it; if there's less friction this way, then that's fine.

Reply:
Oh ok...I know the `map` class in ES6 but not quite familiar with how the corresponding TS types work.